fulp_modules/features/antagonists/bloodsuckers/code/monster_hunter/monsterhunter_datum.dm

```diff
@@ -37,44 +41,79 @@
 	//Give Monster Hunter powers
 	trackvamp.Grant(owner.current)
 	fortitude.Grant(owner.current)
+	//Give Hunter Objective
 	if(give_objectives)
-		//Give Hunter Objective
-		var/datum/objective/bloodsucker/monsterhunter/monsterhunter_objective = new
-		monsterhunter_objective.owner = owner
-		objectives += monsterhunter_objective
-		//Give Theft Objective
-		var/datum/objective/steal/steal_objective = new
-		steal_objective.owner = owner
-		steal_objective.find_target()
-		objectives += steal_objective
+		find_monster_targets()
+	var/datum/map_template/wonderland/wonder = new()
+	if(!wonder.load_new_z())
+		message_admins("The wonderland failed to load.")
+		CRASH("Failed to initialize wonderland!")
 
 	//Teach Stake crafting
 	owner.teach_crafting_recipe(/datum/crafting_recipe/hardened_stake)
 	owner.teach_crafting_recipe(/datum/crafting_recipe/silver_stake)
+	var/mob/living/carbon/human/killer = owner.current
+	var/datum/brain_trauma/special/rabbit_hole/disease = new
+	killer.gain_trauma(disease)
+	sickness = disease
+	var/mob/living/carbon/criminal = owner.current
+	var/obj/item/rabbit_locator/card = new(criminal,src)
+	var/list/slots = list ("backpack" = ITEM_SLOT_BACKPACK)
+	criminal.equip_in_one_of_slots(card, slots)
+	var/obj/item/hunting_contract/contract = new(criminal,src)
+	criminal.equip_in_one_of_slots(contract, slots)
+	RegisterSignal(src, GAIN_INSIGHT, .proc/insight_gained)
+	RegisterSignal(src, BEASTIFY, .proc/turn_beast)
```

Comment:
Signals should at the very least start with COMSIG, you also should generally try to name it after where the signal is. (``BEASTIFY`` doesn't really tell me when this signal is sent)